src/reason_pprint_ast.ml

```diff
@@ -691,6 +691,16 @@ let default = new Pprintast.printer ()
 type funcReturnStyle =
   | ReturnValOnSameLine
 
+let rec detectJSXComponent e attributes =
+  match (e, attributes) with
+    | (Pexp_ident loc, ({txt = "JSX"; loc = _}, PStr []) :: tail) ->
+        let moduleNameList = List.rev (List.tl (List.rev (Longident.flatten loc.txt))) in
+        if List.length moduleNameList > 0 then
```

Comment:
No actually not, I use separate parts of the identifier for both blocks. First part uses everything but the `createElement` part (so the last list item) while the second part uses the last list item which can only be one (which would be a function call).